scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,862 @@
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new ClosureEvaluationException(EXCEPTION_WHEN_EVALUATING_CLOSURE + StackTraceUtil.getStackTrace(e));
+        }
+        return answer;
+    }
+
+    @Override
+    public void conditionalReplaceAll(String originator, TaskId taskid, String channel, String predicate,
+            String thenRemappingFunction) throws InvalidChannelException, CompilationException, IOException {
+        String currentKey = null;
+        try {
+            Channel chosenChannel = getChannel(channel);
+            for (Map.Entry<String, Serializable> entry : chosenChannel.entrySet()) {
+                currentKey = entry.getKey();
+                Serializable value = chosenChannel.get(currentKey);
+                if (evaluateClosure(predicate, BiPredicate.class).test(currentKey, value)) {
+
+                    Serializable answer = chosenChannel.compute(currentKey,
+                                                                evaluateClosure(thenRemappingFunction,
+                                                                                BiFunction.class));
+                    logWithContext(taskid,
+                                   channel,
+                                   "Run conditionalReplaceAll" + ON_KEY + QUOTE + currentKey + QUOTE + RESULT_IS +
+                                            answer,
+                                   Level.DEBUG);
+                }
+            }
+            commitIfNeeded(channel);
+
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new ClosureEvaluationException(EXCEPTION_WHEN_EVALUATING_CLOSURE + StackTraceUtil.getStackTrace(e));
+        }
+    }
+
+    @Override
+    public void conditionalReplaceAll(String originator, TaskId taskid, String channel, String predicate,
+            String thenRemappingFunction, String elseRemappingFunction)
+            throws InvalidChannelException, CompilationException, IOException {
+        String currentKey = null;
+        try {
+            Channel chosenChannel = getChannel(channel);
+            for (Map.Entry<String, Serializable> entry : chosenChannel.entrySet()) {
+                currentKey = entry.getKey();
+                Serializable value = chosenChannel.get(currentKey);
+                if (evaluateClosure(predicate, BiPredicate.class).test(currentKey, value)) {
+
+                    Serializable answer = chosenChannel.compute(currentKey,
+                                                                (BiFunction<? super String, ? super Serializable, ? extends Serializable>) evaluateClosure(thenRemappingFunction,
+                                                                                                                                                           BiFunction.class));
```

Comment:
As mentioned before, we could also have in general a handler function that takes the action to perform and that always logs at the end, instead of having to call each time `logWithContext`.